src/secure-dfu.ts

```diff
@@ -435,30 +435,25 @@ export class SecureDfu extends EventDispatcher {
      * @param uuids Optional alternative uuids for control, packet or button
      * @returns Promise containing the device if it is still on a valid state
      */
-    public setDfuMode(device: BluetoothDevice, uuids?: { service?: number|string, control?: number|string, packet?: number|string, button?: number|string}): Promise<BluetoothDevice> {
-        uuids = uuids || {};
-        const serviceId = uuids.service || SecureDfu.SERVICE_UUID;
-        const controlId = uuids.control || CONTROL_UUID;
-        const packetId = uuids.packet || PACKET_UUID;
-        const buttonId = uuids.button || BUTTON_UUID;
-        return this.gattConnect(device, serviceId)
+    public setDfuMode(device: BluetoothDevice, uuids: UuidOptions = this.DEFAULT_UUIDS): Promise<BluetoothDevice> {
```

Comment:
I like this approach, but it won't fill in the blanks if you only supply a few of the uuids.

You could make them all mandatory in the interface so it's an 'all or nothing' approach or do:

```typescript
public setDfuMode(device: BluetoothDevice, uuids: UuidOptions = this.DEFAULT_UUIDS): Promise<BluetoothDevice> {
    uuids = {
        ...this.DEFAULT_UUIDS,
        ...uuids
    };
    ...
```